### ISSUE TITLE
config.eval_batch_size and config.batch_size? how exactly calculate accuracy in validation dataset?

### ISSUE THREAD

**Issue by sharpstill** [zhongwen/RAM#5]
I notice there is two batch_size in the calculate validation set accuracy ?
 batch_size = 32
eval_batch_size = 50
I can't quite understood the logic inside code below, In my experiment, if the dataset.num_examples < config.eval_batch_size, then steps_per_epoch  = 0, the code below failed!

`            for dataset in [mnist.test]:
                steps_per_epoch = dataset.num_examples // config.eval_batch_size
                correct_cnt = 0
                num_samples = steps_per_epoch * config.batch_size
                loc_net.sampling = True
                for test_step in range(steps_per_epoch):
                    batch_dict = mnist.train.next_batch(
                        config.batch_size,
                        reshape_input=False,
                        read_color_type=cv2.IMREAD_GRAYSCALE)
                    images = batch_dict["imgs"]
                    labels = batch_dict["labels"]
                    # for ii, _img in enumerate(images):  # FIXME !!!
                    #     images[ii] = local_binary_pattern(_img, 8, 24)
                    images = images.reshape(
                        images.shape[0],
                        images.shape[1] *
                        images.shape[2])

                    labels_bak = labels
                    # Duplicate M times
                    images = np.tile(images, [config.M, 1])
                    labels = np.tile(labels, [config.M])
                    softmax_val = sess.run(softmax,
                                           feed_dict={
                                               images_ph: images,
                                               labels_ph: labels
                                           })
                    softmax_val = np.reshape(
                        softmax_val, [
                            config.M, -1, config.num_classes])
                    softmax_val = np.mean(softmax_val, 0)
                    pred_labels_val = np.argmax(softmax_val, 1)
                    pred_labels_val = pred_labels_val.flatten()
                    correct_cnt += np.sum(pred_labels_val == labels_bak)
                acc = correct_cnt / num_samples
`
